### PR TITLE
Feat/apply clean arch enhance code quality

### PR DESCRIPTION
**Pull request by mahmoudashrafmohamed**

### CODE REVIEW

**Review by ahoshaiyan — APPROVED** [review]

File: sdk/src/main/java/com/moyasar/android/sdk/presentation/viewmodel/PaymentSheetViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+   ************************/
+  private fun createSaveOnlyToken() {
+    val request = TokenRequest(
+      formValidator.name.value!!,
+      cleanCardNumber,
+      formValidator.cvc.value!!,
+      expiryMonth,
+      expiryYear,
+      true,
+      "https://sdk.moyasar.com"
```
Comment (Fghurayri) [inline sdk/src/main/java/com/moyasar/android/sdk/presentation/viewmodel/PaymentSheetViewModel.kt:121]: If I understand correctly, the value here should be the full RETURN_URL (`https://sdk.moyasar.com/payment/return`) . Can you please confirm?

**Review by Fghurayri — CHANGES_REQUESTED** [review]
Well done @mahmoudashrafmohamed 🙏 

In addition to reviewing my comments, please consider the following for our next round of review:

- Can you please add unit & integration tests.
- We would like to perform a real payment to ensure the E2E flow is working as expected. Can you please peform this step? If you need help, we can coordinate this point in Teams.

**Comment by Fghurayri** [comment]
Excellent job! 👏 @mahmoudashrafmohamed 

Let's connect in Teams to perform a real E2E prior merging this PR! 🚀

**Review by Fghurayri — APPROVED** [review]
We've successfully performed a real payment on Staging (moyasar_cc_qn3cJ5EqLMNgrRUP6imCi6S) and everything worked as expected, alhamdillah. Thank you @mahmoudashrafmohamed 

@ahoshaiyan we are good to go with the release!

**Pull request merged**